torch/_inductor/pattern_matcher.py

```diff
@@ -1080,56 +1080,64 @@ def check_fn(match: Match):
                         ):
                             sym_args.append(v)
 
-            if sym_args:
-                # AOT Autograd and make fx will dedupe symbolic shape size
-                # accesses of sym ints that appear as inputs
-                # We don't want the sym_size uses to interfere with pattern matching
-                # so we provide them as inputs.
-                # Later, when we actually do the replacement, the symbolic shape
-                # sizes will get re-traced and added to the graph.
-
-                def search_fn_new(*args_new):
-                    return search_fn(*args_new[len(args_new) - len(args) :])
-
-                try:
-                    specific_graph = trace_fn(search_fn_new, sym_args + args)
-                except RuntimeError as e:
-                    log_trace_failure(search_fn, e)
-                    return False
-
-                # correct argnames in the graph
-                sym_arg_names = []
-                for i, placeholder in zip(
-                    range(len(sym_args) + len(args)),
-                    specific_graph.graph.nodes,
-                ):
-                    if i < len(sym_args):
-                        sym_arg_names.append(placeholder.target)
-                        continue
+            # If we were given a pre-traced pattern then use that instead of
+            # retracing. Note that this means the pattern has to be independent
+            # of its args.
+            specific_pattern = search_fn_pattern
```

Comment:
Are there any cases today where this wouldn't be true 🤔  We do have some patterns with `Ignored()` but that's not an issue here since the same pattern with `Ignored()` is generated regardless of the inputs..

Reply:
The misc patterns are this way today. They don't serialize cleanly (I think they depend on their args) so they don't have predefined patterns (they use register_replacement instead of gen_register_replacement)